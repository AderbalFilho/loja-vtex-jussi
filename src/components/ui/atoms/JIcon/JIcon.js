import React from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import searchIcon from '@assets/icons/search.svg';
import shoppingCartIcon from '@assets/icons/shopping-cart.svg';

function JIcon({ classes, icon, ...rest }) {
  const className = classNames('j-icon', classes);
  const iconsDictionary = {
    search: {
      accessibilityText: 'buscar',
      icon: searchIcon,
    },
    shoppingCart: {
      accessibilityText: 'carrinho de compras',
      icon: shoppingCartIcon,
    },
  };

  return (
    <img
      className={className}
      src={icon && iconsDictionary[icon].icon}
      alt={icon && iconsDictionary[icon].accessibilityText}
      {...rest}
    />
  );
}

JIcon.propTypes = {
  classes: PropTypes.string,
  icon: PropTypes.oneOf(['search', 'shoppingCart']).isRequired,
};

export default JIcon;
