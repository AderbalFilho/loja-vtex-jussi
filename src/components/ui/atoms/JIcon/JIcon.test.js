import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import JIcon from './JIcon';

describe('JIcon', () => {
  describe('With icon search', () => {
    it('matches snapshot', () => {
      const testRenderer = TestRenderer.create(<JIcon icon="search" />);
      const jsonComponent = testRenderer.toJSON();

      expect(jsonComponent).toMatchSnapshot();
    });

    it('renders right alt', () => {
      const { container } = render(<JIcon icon="search" />);
      const icon = container.querySelector('.j-icon');

      expect(icon.alt).toContain('buscar');
    });
  });

  describe('With shopping cart search', () => {
    it('matches snapshot', () => {
      const testRenderer = TestRenderer.create(<JIcon icon="shoppingCart" />);
      const jsonComponent = testRenderer.toJSON();

      expect(jsonComponent).toMatchSnapshot();
    });

    it('renders right alt', () => {
      const { container } = render(<JIcon icon="shoppingCart" />);
      const icon = container.querySelector('.j-icon');

      expect(icon.alt).toContain('carrinho de compras');
    });
  });

  it('renders classes', () => {
    const { container } = render(<JIcon classes="icon-search" icon="search" />);
    const icon = container.querySelector('.j-icon');

    expect(icon.classList).toContain('icon-search');
  });
});
