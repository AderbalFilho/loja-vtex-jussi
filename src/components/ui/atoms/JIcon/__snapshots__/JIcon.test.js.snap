// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JIcon With icon search matches snapshot 1`] = `
<img
  alt="buscar"
  className="j-icon"
  src=""
/>
`;

exports[`JIcon With shopping cart search matches snapshot 1`] = `
<img
  alt="carrinho de compras"
  className="j-icon"
  src=""
/>
`;
